tests/validator/validator.cpp

```diff
@@ -608,20 +605,67 @@ TEST(Validator, invalidMathMLVariables)
     EXPECT_EQ_ERRORS(expectedErrors, v);
 }
 
+TEST(Validator, invalidSimpleMathmlCellMLUnits)
+{
+    const std::string math =
+        "<math  xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\"><apply><bvar><ci cellml:units=\"dimensionless\">B</ci></bvar></apply></math>";
+    const std::vector<std::string> expectedErrors = {
+        "CellML identifiers must contain one or more basic Latin alphabetic characters.",
+        "Model does not have a valid name attribute.",
+        "CellML identifiers must contain one or more basic Latin alphabetic characters.",
+        "Component does not have a valid name attribute.",
+        "MathML ci element has the child text 'B' which does not correspond with any variable names present in component ''.",
+        "W3C MathML DTD error: No declaration for attribute units of element ci.",
+    };
+    libcellml::Validator v;
+    libcellml::ModelPtr m = libcellml::Model::create();
+    libcellml::ComponentPtr c = libcellml::Component::create();
+
+    c->setMath(math);
+    m->addComponent(c);
+
+    v.validateModel(m);
+    EXPECT_EQ_ERRORS(expectedErrors, v);
+}
+
+TEST(Validator, invalidMathmlCellMLNsOnNode)
+{
+    const std::string math =
+        "<math  xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\"><apply><cellml:bvar><ci cellml:units=\"dimensionless\">B</ci></cellml:bvar></apply></math>";
```

Comment:
I wish this test was based on a physical CellML file rather than inline code like here, which makes it more difficult to assess/review/understand/etc.

Comment also valid elsewhere.

Reply:
whereas I like to see them inline to make it easy to see everything in one place :)

At least for these small test models.

Reply:
Sure, in this particular case, it was more about the lack of indentation and new lines. When it's some (short) CellML code that is indented and has new lines, then it's fine.

Reply:
I much prefer cellml inline but with indents for code that's not a real model.  This means that it can be commented so we can follow the tests' purposes more easily, and saves hunting around for files when they only contain a couple of lines.  Indents are needed though ... :)